lighthouse-core/computed/metrics/cumulative-layout-shift-all-frames.js

```diff
@@ -5,10 +5,38 @@
  */
 'use strict';
 
+/** @typedef {Omit<LH.TraceEvent, 'name'|'args'> & {name: 'LayoutShift', args: {data: {score: number, weighted_score_delta: number, had_recent_input: boolean}}}} LayoutShiftEvent */
+
 const makeComputedArtifact = require('../computed-artifact.js');
 const TraceOfTab = require('../trace-of-tab.js');
+const LHError = require('../../lib/lh-error.js');
 
 class CumulativeLayoutShiftAllFrames {
+  /**
+   * @param {LH.TraceEvent} event
+   * @return {event is LayoutShiftEvent}
+   */
+  static isLayoutShiftEvent(event) {
+    if (
+      event.name !== 'LayoutShift' ||
+      !event.args ||
+      !event.args.data ||
+      !event.args.data.score ||
+      event.args.data.had_recent_input
+    ) return false;
+
+    // Weighted score was added to the trace in m89:
+    // https://bugs.chromium.org/p/chromium/issues/detail?id=1173139
+    if (typeof event.args.data.weighted_score_delta !== 'number') {
+      throw new LHError(
```

Comment:
should this fallback to score? seems like there's still some value in seeing if the all frames version and main-frame version are different. I understand the difficulty of figuring out which CLS values are valid, but seems like we're going to have to do that anyway with a mix of LH versions in the same month.

Reply:
I'm good with doing a fallback and replacing this with a log warning for now.

I'm just worried that the fallback doesn't make sense in the long run. Once we get through the churn of different LH and Chrome versions I would like to remove the fallback.

Reply:
SG, can throw a `FIXME: remove after XX` on there :)